src/include/distributed/metadata_cache.h

```diff
@@ -49,7 +50,15 @@ typedef struct
 	int shardIntervalArrayLength;
 	ShardInterval **sortedShardIntervalArray;
 
-	FmgrInfo *shardIntervalCompareFunction; /* NULL if no shard intervals exist */
+	/* comparator for partition column's type, NULL if DISTRIBUTE_BY_NONE */
+	FmgrInfo *shardValueCompareFunction;
```

Comment:
This suggests there is a "shard value". Maybe `partitionColumnValueCompareFunction` or `partitionValueCompareFunction`.